src/components/Menu/ActionInsertLink.vue

```diff
@@ -232,11 +232,7 @@ export default {
 			getLinkWithPicker(null, true)
 				.then(link => {
 					const chain = this.$editor.chain()
-					if (this.$editor.view.state?.selection.empty) {
-						chain.focus().insertContent(link + ' ').run()
-					} else {
-						chain.setLink({ href: link }).focus().run()
-					}
+					chain.focus().insertPreview(link).run()
```

Comment:
I'd vote to only do this when the selection is still empty

Reply:
Not sure if that's what users expect. If you insert a link via smart picker, you already see the preview in the picker. So it's weird to then only insert a text link in my opinion.

An alternative I've considered is to always insert the preview link *after* the paragraph if the cursor is inside a paragraph (and not at beginning or end).

Reply:
After discussing different scenarios with @max-nextcloud I agree with your suggestion @juliushaertl and implemented it this way now. See the described behaviour above in the PR description.